lib/y2logsstep.pm

```diff
@@ -262,6 +273,27 @@ sub post_fail_hook {
     assert_script_run 'df > /tmp/df.txt';
     upload_logs '/tmp/df.txt';
 
+    $self->save_and_upload_log('pstree',  '/tmp/pstree');
+    $self->save_and_upload_log('ps auxf', '/tmp/ps_auxf');
+
+    # Collect yast2 installer trace
+    if (!script_run(qq{ps -eo pid,comm | grep -i [y]2start | cut -f 2 -d " " > /dev/$serialdev}, 0)) {
+        chomp(my $installer_pid = wait_serial(qr/^[\d{4}]/, 10));
+        my $trace_timeout = 120;
+        my $strace_ret = script_run("timeout $trace_timeout strace -f -o /installer_trace.log -tt -p $installer_pid", ($trace_timeout + 5));
```

Comment:
I guess it is more stable to parse the pid within the shell session instead of forwarding to the test context

Reply:
I see, then it makes sense of course